compyle/profile.py

```diff
@@ -0,0 +1,100 @@
+""" Utils for profiling kernels
+"""
+
+import time
+from contextlib import contextmanager
+from collections import defaultdict
+from .config import get_config
+
+
+_profile_info = defaultdict(lambda: {'calls': 0, 'time': 0})
+
+
+def record_profile(name, time):
+    global _profile_info
+    _profile_info[name]['time'] += time
+    _profile_info[name]['calls'] += 1
+
+
+@contextmanager
+def profile(name):
+    start = time.time()
+    yield start
+    end = time.time()
+    record_profile(name, end - start)
+
+
+def profile_method(method):
+    def wrapper(*args, **kwargs):
+        self = args[0] if len(args) else None
+        with profile(getattr(self, "name", method.__name__)):
+            return method(*args, **kwargs)
+    return wrapper
+
+
+def print_profile():
+    global _profile_info
+    profile_data = sorted(_profile_info.items(), key=lambda x: x[1]['time'],
+                          reverse=True)
+    if len(_profile_info) == 0:
+        print("No profiling information available")
+        return
+    print("Profiling info:")
+    print("{:<40} {:<10} {:<10}".format('Function', 'N calls', 'Time'))
+    tot_time = 0
+    for kernel, data in profile_data:
+        print("{:<40} {:<10} {:<10}".format(
+            kernel,
+            data['calls'],
+            data['time']))
+        tot_time += data['time']
+    print("Total profiled time: %g secs" % tot_time)
+
+
+def profile_kernel(kernel, name, backend=None):
```

Comment:
Some docstring with an example or additional documentation would be useful, there are now many public functions and it isn't clear what is to be used.  I have myself forgotten!